### PR TITLE
Refactor dockerfile to use private cache sharing

### PR DESCRIPTION
**Pull request by jLopezbarb**
# Proposed changes

Fixes DEV-683
Allow to run multiple cache for tests writing in the same cache

## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines

### CODE REVIEW

File: pkg/remote/remote.go [hunk 1]
```diff
@@ -102,7 +102,7 @@ ENV {{$key}}={{$val}}
 {{ end }}
 
 RUN \
-  {{range $key, $path := .Caches }}--mount=type=cache,target={{$path}} {{end}}\
+  {{range $key, $path := .Caches }}--mount=type=cache,target={{$path}},sharing=private {{end}}\
```
[the remark this answers is not in the document]
Comment (jLopezbarb) [reply pkg/remote/remote.go]: Added on https://github.com/okteto/okteto/pull/4463/commits/3bcd5be69086aa66483b058180ff2c1bb09028b4

**Review by ifbyol — COMMENTED** [review]
Would it make sense to do it at manifest level instead of a command flag? If we go with this approach first, I'd include also the flag in the github action to customize it

**Comment by jLopezbarb** [comment]
> Would it make sense to do it at manifest level instead of a command flag? If we go with this approach first, I'd include also the flag in the github action to customize it

If we introduce this at the manifest level, we'd need to either split or modify the cache structure to include both the sharing method and caches, like this:

```yaml
test:
  caches:
    shared: sharing
    caches:
    - /root/go
 ```
Since all caches must be consistent for proper behavior, I opted for a command flag instead. This avoids manifest changes and provides flexibility (also the naming for the cache is going to give some headache). I agree that if we take this approach, we should also add it to the GitHub Action for customization.

Thoughts?

**Comment by pchico83** [comment]
> > > Would it make sense to do it at manifest level instead of a command flag? If we go with this approach first, I'd include also the flag in the github action to customize it
> > 
> > 
> > If we introduce this at the manifest level, we'd need to either split or modify the cache structure to include both the sharing method and caches, like this:
> > ```yaml
> > test:
> >   caches:
> >     shared: sharing
> >     caches:
> >     - /root/go
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Since all caches must be consistent for proper behavior, I opted for a command flag instead. This avoids manifest changes and provides flexibility (also the naming for the cache is going to give some headache). I agree that if we take this approach, we should also add it to the GitHub Action for customization.
> > Thoughts?
> 
> You might want to execute 2 different tests with different cache strategies. How would you do that? Also, if I want to change the cache for a test always, I need to remember to specify it always when I execute that test.
> 
> Ask in #product, but I'd split it into 2 things:
> 
> * Change the default cache for the tests to internal. As it is now, it will generate issues mainly with any test run. This is a change with more priority
> * Make it configurable, so anyone could change it if needed. In this case, we should support it at manifest level to allow to configure each test with a cache type. This can go in another release, as the problematic one is the one configured currently

+1 for changing the default and creating another issue to make it configurable. Changing the default is blocking parallel executions. Making it configurable is great, but it ahs less priority and we can have more time to define the syntax at the yaml file level

**Review by ifbyol — APPROVED** [review]

**Pull request merged**